### PR TITLE
Move general plot options to sidebar

### LINKED ISSUE #13569: Add `customization`-widget functionality to `Plot controls`-panel
We could extract the most important/used features from the existing `Customization`-widget to the `Plot controls`-panel, as well as moving all plot tick boxes (extended plot info and log scale) to the same place. 

Screenshots of non-functioning draft:

Image

Image

### LINKED ISSUE #13963: Move log scale checkbox to plot controls panel
We want to move the log-scale checkbox to the right side of the screen.
This will free up screen estate for plots that are eligible for showing log scale.

Also, there is reserved space for the log-scale checkbox and this needs to be removed to allow the plot to increase its size.

Image

Relates to #13769 

The checkbox could be positioned at the bottom to the right, similar to examples found in the screenshots here;
https://github.com/equinor/ert/issues/13569

### PR DESCRIPTION
**Pull request by MagnusSletten**
**Issue**
Resolves #13963 
Resolves #13569

**Approach**
I split the side bar options into two categories, general options and plot specific options. They have their own segments so it's easy to navigate. Added a simple file for the general plotting options and removed the corresponding information from default_customization_view. 

Change in behaviour: The visibility of the plot option checkboxes now depend on their relevance. If the checkboxes are incompatible with the current plot/data then they are not displayed. 

image

image

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'just rapid-tests'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Add backport label to latest release (format: 'backport release-branch-name')

### CODE REVIEW

**Comment by ertomatic** [comment]
Screenshots differ from baselines. A baseline update PR has been prepared: https://github.com/equinor/ert-testdata/pull/47

File: src/ert/gui/plotting/widgets/plot_controls/general_options.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        self._toggle_history.setChecked(True)
+        self._toggle_history.stateChanged.connect(connection_point)
+
+        self._toggle_observations = QCheckBox("Observations")
+        self._toggle_observations.setChecked(True)
+        self._toggle_observations.stateChanged.connect(connection_point)
+
+        self._toggle_log_scale = QCheckBox("Log scale")
+        self._toggle_log_scale.setChecked(False)
+        self._toggle_log_scale.stateChanged.connect(connection_point)
```
[the remark this answers is not in the document]
Comment (MagnusSletten) [reply src/ert/gui/plotting/widgets/plot_controls/general_options.py]: Fixed

File: src/ert/gui/plotting/customization_dialog/default_customization_view.py [hunk 2]
```diff
@@ -82,11 +60,5 @@ def revert_customization(self, plot_config: "PlotConfig") -> None:
...
-        self.x_label = plot_config.x_label()
-        self.y_label = plot_config.y_label()
-
-        self.legend = plot_config.is_legend_enabled()
-        self.grid = plot_config.is_grid_enabled()
-        if not self.is_everest:
-            self.history = plot_config.is_history_enabled()
-            self.observations = plot_config.is_observations_enabled()
+        self.x_label = plot_config.x_label() or ""
+        self.y_label = plot_config.y_label() or ""
```
[the remark this answers is not in the document]
Comment (MagnusSletten) [reply src/ert/gui/plotting/customization_dialog/default_customization_view.py]: Fixed

File: tests/ert/ui_tests/gui/test_main_window.py [hunk 3]
```diff
@@ -406,15 +406,9 @@ def get_log_checkbox():
...
-        qtbot.waitUntil(lambda: get_log_checkbox() is not None, timeout=2000)
         cb = get_log_checkbox()
-
-        # wait until it becomes visible
-        qtbot.waitUntil(
-            lambda: cb.isVisibleTo(plot_window._central_tab.currentWidget()) is True,
-            timeout=2000,
-        )
+        assert cb is not None
+        assert not cb.isVisible()
```
[the remark this answers is not in the document]
Comment (MagnusSletten) [reply tests/ert/ui_tests/gui/test_main_window.py:423]: FIxed 👍

File: src/ert/gui/plotting/plot_window.py [hunk 4]
```diff
@@ -355,6 +366,32 @@ def __init__(
...
+                a = key_def.parameter
+                if isinstance(a, GenKwConfig) and isinstance(
+                    a.distribution, ConstSettings
+                ):
+                    self._general_options.set_log_visible(False)
+                    return
+            self._general_options.set_log_visible(True)
+        else:
+            self._general_options.set_log_visible(False)
+
```
Comment (eilskra) [inline src/ert/gui/plotting/plot_window.py]: I dont think this is needed. The [`log_scale_valid_values`](https://github.com/equinor/ert/blob/main/src/ert/gui/plotting/plot_window.py#L503) have already checked for negative values and constant distributions. 

I think visibility could simplified to:
`self._general_options.set_log_visible(log_scale_valid_values and selected_tab in { HISTOGRAM, DISTRIBUTION, GUASSIAN_KDE }`

File: src/ert/gui/plotting/plot_window.py [hunk 5]
```diff
@@ -549,6 +586,16 @@ def fetch_data(
             plot_config = PlotConfig.create_copy(
                 self._plot_customizer.get_plot_config()
             )
+            plot_config.set_legend_enabled(self._general_options.legend_checkbox_state)
+            plot_config.set_grid_enabled(self._general_options.grid_checkbox_state)
+            if not self.is_everest:
+                plot_config.set_history_enabled(
+                    self._general_options.history_checkbox_state
+                )
+                plot_config.set_observations_enabled(
```
Comment (eilskra) [inline src/ert/gui/plotting/plot_window.py:582]: Not all ERT data contains observations (in the GUI, its the ones marked with a yellow highlight), so we need to check that before setting this as true. Checking how the `MisfitsPlot` becomes [available](https://github.com/equinor/ert/blob/main/src/ert/gui/plotting/plot_window.py#L706) can be a good pointer to how to check this

File: src/ert/gui/plotting/plot_window.py [hunk 6]
```diff
@@ -549,6 +586,16 @@ def fetch_data(
             plot_config = PlotConfig.create_copy(
                 self._plot_customizer.get_plot_config()
             )
+            plot_config.set_legend_enabled(self._general_options.legend_checkbox_state)
+            plot_config.set_grid_enabled(self._general_options.grid_checkbox_state)
+            if not self.is_everest:
+                plot_config.set_history_enabled(
```
Comment (eilskra) [inline src/ert/gui/plotting/plot_window.py:578]: Similar as comment about observations. Think its determined [here](https://github.com/equinor/ert/blob/main/src/ert/gui/plotting/plot_window.py#L568)

File: src/ert/gui/plotting/utils/plot_config.py
Comment (eilskra) [inline src/ert/gui/plotting/utils/plot_config.py:1]: Unsure if needed since `PlotTools.finalizePlots()` always sets a default label. In general we could prob. simplify and improve the `plotting/utils`-a lot, but should be its own PR

File: tests/ert/unit_tests/gui/tools/plot/test_plot_window.py [hunk 7]
```diff
@@ -298,6 +301,68 @@ def get_x_axis_scale() -> str:
...
+        index_type=None,
+        metadata={"data_origin": "GEN_KW"},
+        observations=False,
+        dimensionality=1,
+        parameter=GenKwConfig(
+            name="gen_kw_nonconst",
+            distribution={"name": "uniform", "min": 0.0, "max": 1.0},
+        ),
+    )
+    const_key_def = PlotApiKeyDefinition(
```
Comment (eilskra) [inline tests/ert/unit_tests/gui/tools/plot/test_plot_window.py]: Could we also parametrize these? Maybe also testing for negative values could be good

**Review by eilskra — COMMENTED** [review]
Just quickly checked, looking good so far 👍 Added a few comments

File: src/ert/gui/plotting/customization_dialog/default_customization_view.py
Comment (eilskra) [inline src/ert/gui/plotting/customization_dialog/default_customization_view.py:1]: Could we move the title and label edits to `Style` tab or can these fit in the sidebar too? If either solutions work, we could remove `default_customization_view.py`

Comment (MagnusSletten) [reply src/ert/gui/plotting/customization_dialog/default_customization_view.py:1]: There was a suggestion about simply moving these to the plotted window itself so you can click the axis label to change it.

Comment (eilskra) [reply src/ert/gui/plotting/customization_dialog/default_customization_view.py:1]: Would suggest clicking axis label as a complimentary feature, rather than the only one. Think it could prove to be a bit difficult for some users, so feel like having a non "shortcut" way is good 👍

Comment (MagnusSletten) [reply src/ert/gui/plotting/customization_dialog/default_customization_view.py:1]: I agree! I'll do that

File: src/ert/gui/plotting/utils/plot_config.py
Comment (MagnusSletten) [reply src/ert/gui/plotting/utils/plot_config.py:1]: The change is just to avoid the IDE complaining about type hints, the current behaviour (before this PR) also can insert None here.  It being Nullable is important for: 
https://github.com/equinor/ert/blob/137d9f2e8f0f1a328e59ae10bfe2565209decbd3/src/ert/gui/plotting/utils/plot_tools.py#L171

File: src/ert/gui/plotting/customization_dialog/default_customization_view.py
Comment (MagnusSletten) [reply src/ert/gui/plotting/customization_dialog/default_customization_view.py:1]: Done

File: tests/ert/unit_tests/gui/tools/plot/test_plot_window.py (hunk 7 above)
Comment (MagnusSletten) [reply tests/ert/unit_tests/gui/tools/plot/test_plot_window.py]: Done

File: src/ert/gui/plotting/widgets/plot_controls/general_options.py [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+from ert.gui.plotting.utils.qt_creator import create_group_box, create_group_layout
+
+
+class GeneralOptions:
+    def __init__(
+        self, connection_point: Callable[..., object], *, is_everest: bool
+    ) -> None:
+        def refresh_plot(_state: int) -> None:
+            connection_point()
```
Comment (eilskra) [inline src/ert/gui/plotting/widgets/plot_controls/general_options.py]: Is this needed? `MisfitOptions` just used `connection_point` directly

File: tests/ert/ui_tests/gui/test_plot_customization.py [hunk 9]
```diff
@@ -15,8 +15,8 @@ def test_that_first_tab_is_not_logged_when_opening_customize_plot_dialog(qtbot,
     )
 
     plot = CustomizePlotDialog(title="Test Plot", parent=None, key_defs=[])
-    plot.add_tab("general", "General", DefaultCustomizationView())
-    plot.add_tab("style", "Style", DefaultCustomizationView())
+    plot.add_tab("general", "General", StyleCustomizationView())
+    plot.add_tab("style", "Style", StyleCustomizationView())
```
Comment (eilskra) [inline tests/ert/ui_tests/gui/test_plot_customization.py]: Do we need `general`, is this not duplicating `StyleCustomizationView()`?

File: tests/ert/unit_tests/gui/tools/plot/test_plot_window.py [hunk 10]
```diff
@@ -298,6 +298,53 @@ def get_x_axis_scale() -> str:
     assert get_x_axis_scale() == "log", "Plot scale should still be log"
 
 
+def test_that_unavailable_history_and_observations_checkboxes_are_hidden(
```
Comment (eilskra) [inline tests/ert/unit_tests/gui/tools/plot/test_plot_window.py]: Could we also test that it becomes available? 

This could be done in same test, but parameterized with conditions where they should/should not show or in another test

File: src/ert/gui/plotting/plot_window.py [hunk 11]
```diff
@@ -364,6 +364,16 @@ def __init__(
             if self.getSelectedKey() is None:
                 self._show_no_data_message()
 
+    def _sync_log_scale_option(
```
Comment (eilskra) [inline src/ert/gui/plotting/plot_window.py]: should this maybe be part of `GeneralOptions`?

Comment (eilskra) [reply src/ert/gui/plotting/plot_window.py]: I would just in-line it tbh. I dont think it needs the extra lines

File: src/ert/gui/plotting/plot_window.py [hunk 12]
```diff
@@ -396,6 +406,27 @@ def logPlotTabUsage(self, tab_name: str, *, default: bool = False) -> None:
     def layerIndexChanged(self, index: int | None) -> None:
         self.updatePlot(index)
 
+    def _log_scale_valid_values(
```
Comment (eilskra) [inline src/ert/gui/plotting/plot_window.py]: Unsure if this needs its own method. Adds additional lines and is only used by the log scale button

File: tests/ert/unit_tests/gui/tools/plot/test_plot_window.py [hunk 13]
```diff
@@ -344,6 +348,207 @@ def test_that_unavailable_history_and_observations_checkboxes_are_hidden(
     assert not observations_checkbox.isVisible()
 
 
+def test_that_log_scale_state_is_preserved_when_switching_plot_tabs(
```
Comment (eilskra) [inline tests/ert/unit_tests/gui/tools/plot/test_plot_window.py:593]: Should we maybe add these tests and additional `GeneralOptions`tests to `test_general_options.py`?

Comment (eilskra) [reply tests/ert/unit_tests/gui/tools/plot/test_plot_window.py:593]: We should prob test that the checkboxes changes plot behavior as expected, e.g if legend is checked `axes` contains legend and vice versa

File: src/ert/gui/plotting/widgets/plot_controls/general_options.py (hunk 8 above)
Comment (MagnusSletten) [reply src/ert/gui/plotting/widgets/plot_controls/general_options.py]: Done! Now it's using connection point which emits different integers depending on the state, but this is not a problem since layer value is not relevant for plots with general options.

File: tests/ert/ui_tests/gui/test_plot_customization.py (hunk 9 above)
Comment (MagnusSletten) [reply tests/ert/ui_tests/gui/test_plot_customization.py]: Yeah it's just a test for switching tabs as it's related to logging so having a duplicate here is alright. It just requires two tabs, but I changed it to style and statistics.

File: tests/ert/unit_tests/gui/tools/plot/test_plot_window.py (hunk 10 above)
Comment (MagnusSletten) [reply tests/ert/unit_tests/gui/tools/plot/test_plot_window.py]: Added several tests on this

File: src/ert/gui/plotting/plot_window.py (hunk 12 above)
Comment (MagnusSletten) [reply src/ert/gui/plotting/plot_window.py]: Fixed

File: tests/ert/unit_tests/gui/plotting/widgets/test_general_options.py [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+):
+    options = GeneralOptions(Mock(), is_everest=False)
+    qtbot.addWidget(options.get_widget())
+    options.get_widget().show()
+
+    options.set_history_visible(history_visible)
+    options.set_observations_visible(observations_visible)
+
+    assert options._toggle_history.isVisible() is history_visible
+    assert options._toggle_observations.isVisible() is observations_visible
```
Comment (eilskra) [inline tests/ert/unit_tests/gui/plotting/widgets/test_general_options.py]: Recommend doing this through `qtbot.mouseClick()`, see pattern in https://github.com/equinor/ert/blob/main/tests/ert/unit_tests/gui/plotting/widgets/test_everest_controls_plot_options.py#L20

File: tests/ert/ui_tests/gui/test_main_window.py
Comment (eilskra) [inline tests/ert/ui_tests/gui/test_main_window.py:1]: What about `GeneralOptions` should they not be showing here?

File: src/ert/gui/plotting/widgets/plot_controls/general_options.py [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+    QGroupBox,
+    QWidget,
+)
+
+from ert.gui.plotting.utils.qt_creator import create_group_box, create_group_layout
+
+logger = logging.getLogger(__name__)
+
+
+class GeneralOptions:
```
Comment (eilskra) [inline src/ert/gui/plotting/widgets/plot_controls/general_options.py]: Should we maybe rename this? Introduce that this is plot related somehow. Let me know what you think

File: src/ert/gui/plotting/widgets/plot_controls/general_options.py [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+        self._toggle_history = QCheckBox("History")
+        self._toggle_history.setChecked(True)
+        self._toggle_history.stateChanged.connect(connection_point)
+
+        self._toggle_observations = QCheckBox("Observations")
+        self._toggle_observations.setChecked(True)
+        self._toggle_observations.stateChanged.connect(connection_point)
+
+        self._toggle_log_scale = QCheckBox("Log scale")
+        self._toggle_log_scale.setObjectName("log_scale_checkbox")
```
Comment (eilskra) [inline src/ert/gui/plotting/widgets/plot_controls/general_options.py:55]: could do this for all `QCheckboxes` if intending to use `findChild()` during testing

File: src/ert/gui/plotting/plot_window.py [hunk 17]
```diff
@@ -355,6 +364,16 @@ def __init__(
...
+    def _sync_log_scale_option(
+        self,
+        selected_tab: str,
+        log_scale_valid_values: bool,
+    ) -> None:
+        self._general_options.set_log_visible(
+            log_scale_valid_values
+            and selected_tab in {HISTOGRAM, DISTRIBUTION, GAUSSIAN_KDE}
+        )
+
```
Comment (eilskra) [inline src/ert/gui/plotting/plot_window.py]: I think I'd inline this to reduce amount of lines needed as its only used once

File: tests/ert/ui_tests/gui/test_main_window.py
Comment (MagnusSletten) [reply tests/ert/ui_tests/gui/test_main_window.py:1]: Fixed!

File: tests/ert/unit_tests/gui/tools/plot/test_plot_window.py (hunk 13 above)
Comment (MagnusSletten) [reply tests/ert/unit_tests/gui/tools/plot/test_plot_window.py:593]: Done

File: src/ert/gui/plotting/plot_window.py (hunk 11 above)
Comment (MagnusSletten) [reply src/ert/gui/plotting/plot_window.py]: Fixed

File: src/ert/gui/plotting/widgets/plot_controls/general_options.py [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+
+        self._toggle_history = QCheckBox("Show History")
+        self._toggle_history.setObjectName("history_checkbox")
+        self._toggle_history.setChecked(True)
+        self._toggle_history.stateChanged.connect(connection_point)
+
+        self._toggle_observations = QCheckBox("Show Observations")
+        self._toggle_observations.setObjectName("observations_checkbox")
+        self._toggle_observations.setChecked(True)
+        self._toggle_observations.stateChanged.connect(connection_point)
```
Comment (eilskra) [inline src/ert/gui/plotting/widgets/plot_controls/general_options.py:52]: We should prob include tooltips for these, `setToolTip()`

File: src/ert/gui/plotting/widgets/plot_controls/general_options.py [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+
+        self._toggle_log_scale = QCheckBox("Log scale")
+        self._toggle_log_scale.setObjectName("log_scale_checkbox")
+        self._toggle_log_scale.setChecked(False)
+        self._toggle_log_scale.setVisible(False)
+
+        self._toggle_log_scale.setToolTip("Toggle data domain to log scale and back")
+
+        def log_log_scale_usage(_checked: bool) -> None:
```
Comment (eilskra) [inline src/ert/gui/plotting/widgets/plot_controls/general_options.py]: Should we add logging for the other options too?

File: tests/ert/unit_tests/gui/tools/plot/test_plot_window.py
Comment (eilskra) [inline tests/ert/unit_tests/gui/tools/plot/test_plot_window.py:1]: I just had a quick thought: 
I see that this is becoming quite large and I'm not sure if this should fall under `plot_window` as this is technically part of the main_window instead, could we make a similar test-file but only for the `PlotControls`-panel relevant tests. 

Could potentially be a different pr, if decide that it fits better in a different PR just resolve comment 👍

**Review by eilskra — COMMENTED** [review]
Just 3 suggestions, just resolve if disagree. Otherwise look goods good 👍

File: tests/ert/unit_tests/gui/tools/plot/test_plot_window.py
Comment (MagnusSletten) [reply tests/ert/unit_tests/gui/tools/plot/test_plot_window.py:1]: I think a separate PR for this would be nice, I can make one right away. A bit easier with the clutter for reviewing as well.

File: src/ert/gui/plotting/widgets/plot_controls/general_options.py (hunk 19 above)
Comment (MagnusSletten) [reply src/ert/gui/plotting/widgets/plot_controls/general_options.py]: Done

File: src/ert/gui/plotting/widgets/plot_controls/general_options.py (hunk 18 above)
Comment (MagnusSletten) [reply src/ert/gui/plotting/widgets/plot_controls/general_options.py:52]: Done

**Review by eilskra — APPROVED** [review]
LGTM! 👍  Good job! 🙌  Remember to merge screenshots prior to rebasing this PR
